src/Avalonia.Controls/Notifications/WindowNotificationManager.cs

```diff
@@ -0,0 +1,170 @@
+            set { SetValue(PositionProperty, value); }
+        }
+
+        public static int GetMaxItems(TopLevel obj)
+        {
+            return obj.GetValue(MaxItemsProperty);
+        }
+
+        public static void SetMaxItems(TopLevel obj, int value)
+        {
+            obj.SetValue(MaxItemsProperty, value);
+        }
+
+        public static readonly AvaloniaProperty<int> MaxItemsProperty =
+          AvaloniaProperty.RegisterAttached<WindowNotificationManager, TopLevel, int>("Position", defaultValue: 5, inherits: true);
+
+        public int MaxItems
+        {
+            get { return GetValue(MaxItemsProperty); }
+            set { SetValue(MaxItemsProperty, value); }
+        }
+
+        public WindowNotificationManager(Window host)
+        {
+            if (VisualChildren.Count != 0)
+            {
+                Install(host);
+            }
+            else
+            {
+                IDisposable stopListening = null;
+
+                stopListening = Observable.FromEventPattern<TemplateAppliedEventArgs>(host, nameof(host.TemplateApplied))
+                    .Subscribe(_ =>
+                    {
+                        stopListening.Dispose();
+
+                        Install(host);
+                    });
+            }
+        }
+
+        static WindowNotificationManager()
+        {
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.TopLeft, ":topleft");
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.TopRight, ":topright");
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.BottomLeft, ":bottomleft");
+            PseudoClass<WindowNotificationManager, NotificationPosition>(PositionProperty, x => x == NotificationPosition.BottomRight, ":bottomright");
+
+            HorizontalAlignmentProperty.OverrideDefaultValue<WindowNotificationManager>(Layout.HorizontalAlignment.Stretch);
+            VerticalAlignmentProperty.OverrideDefaultValue<WindowNotificationManager>(Layout.VerticalAlignment.Stretch);
+        }
+
+        protected override void OnTemplateApplied(TemplateAppliedEventArgs e)
+        {
+            base.OnTemplateApplied(e);
+
+            var itemsControl = e.NameScope.Find<Panel>("PART_Items");
+            _items = itemsControl?.Children;
+        }
+
+        public void Show(INotification content)
+        {
+            Show(content as object);
+        }
+
+        public async void Show(object content)
+        {
+            var notification = content as INotification;
+
+            var notificationControl = new NotificationCard
+            {
+                Content = content
+            };
+
+            if (notification != null)
+            {
+                notificationControl.NotificationClosed += (sender, args) => notification.OnClose?.Invoke();
```

Comment:
Would it not make sense to raise this event in `OnNotificationClosed`?

Reply:
OnNotificationClosed is meant to be the handler. removed.